internals/daemon/daemon.go

```diff
@@ -783,7 +794,14 @@ func New(opts *Options) (*Daemon, error) {
 		httpAddress:         opts.HTTPAddress,
 	}
 
-	ovld, err := overlord.New(opts.Dir, d, opts.ServiceOutput)
+	ManagerGenerators = append(
+		ManagerGenerators,
+		func(o *overlord.Overlord) (overlord.ManagerTag, overlord.StateManager) {
+			commandMgr := cmdstate.NewManager(o.TaskRunner())
+			return CommandManagerTag, commandMgr
+		})
+
+	ovld, err := overlord.New(opts.Dir, d, opts.ServiceOutput, ManagerGenerators)
```

Comment:
Instead of using this scheme to now add all built-in overlord managers as well as external ones:

1. Why do we not leave the overlord code as is today, with your ```stateengine``` tweak to make the ```taskRunner``` special.

2. Add a daemon access method to get to the overlord instance? Then we can do this:

```
func (d *Daemon) Overlord() *overlord.Overlord {
    return d.overlord
}
```

Then in ```internals/cli/cmd_run.go``` equivalent in an importer project, we can do:

```
        d, err := daemon.New(&dopts)
        if err != nil {
                return err
        }
        
        // Add custom managers
        c := customManager.New(d.Overlord().State(), d.Overlord().TaskRunner(), ...)
        d.Overlord().StateEngine().AddManager(c)
        
        if err := d.Init(); err != nil {
                return err
        }

```

Reply:
Yours is a valid way of doing it.
I wanted to avoid exposing the overlord. I don't have a strong preference either way.